src/docs/guides/laravel.md

```diff
@@ -0,0 +1,199 @@
+
+
+## Deploy from the CLI
+
+If you have your Laravel app locally, you can follow these steps:
+
+1. <a href="/guides/cli#installing-the-cli" target="_blank">Install</a> and <a href="/guides/cli#authenticating-with-the-cli" target="_blank">authenticate with the Railway CLI.</a>
+2. Run `railway init` within your Laravel app root directory to create a new project on Railway. 
+    - Follow the steps in the prompt to give your project a name.
+3. Run `railway up` to deploy.
+    - The CLI will now scan, compress and upload our Laravel app files to Railway's backend for deployment.
+    - Your terminal will display real-time logs as your app is being deployed on Railway.
+4. Once the deployment is successful, click on **View logs** on the recent deployment on the dashboard.
+    - You'll see that the server is running. However you'll also see logs prompting you to add your env variables.
+5. Click on the <a href="/overview/the-basics#service-variables">**Variables**</a> section of your service on the Railway dashboard.
+6. Click on **Raw Editor** and add all your app environment variables.
+7. Click on **Deploy** to redeploy your app.
+
+To set up a publicly accessible URL for the service, navigate to the **Networking** section in the [Settings](/overview/the-basics#service-settings) tab of your new service and click on [Generate Domain](/guides/public-networking#railway-provided-domain).
+
+**Note:** The next step shows how to run your Laravel app along with a database, migrations, cron jobs, and workers.
+
+## Set Up Database, Migrations, Crons and Workers
+
+This setup deploys your Laravel app on Railway, ensuring that your database, scheduled tasks (crons), and queue workers are all fully operational.
+
+The deployment structure follows a "majestic monolith" architecture, where the entire Laravel app is managed as a single codebase but split into four separate services on Railway:
+- **App Service**: Handles HTTP requests and user interactions.
+- **Cron Service**: Manages scheduled tasks (e.g., sending emails or running reports).
+- **Worker Service**: Processes background jobs from the queue.
+- **Database Service**: Stores and retrieves your application's data.
+
+<Image src="https://res.cloudinary.com/railway/image/upload/f_auto,q_auto/v1727910244/docs/quick-start/deploy%20architecture.png"
+alt="screenshot of the deploy architecture of the Laravel app"
+layout="responsive"
+width={3118} height={1776} quality={100} />
+_My Majestic Monolith Laravel app_
+
+
+Please follow these steps to get started:
+
+1. Create three bash scripts in the root directory of your Laravel app directory: `run-app.sh`, `run-worker.sh`, and `run-cron.sh`. 
+
+    These scripts will contain the commands needed to deploy and run the app, worker, and cron services for your Laravel app on Railway. 
+    
+    After creating the files, make them executable by running `chmod +x` on each file in the terminal. This will allow your scripts to run properly when executed.
```

Comment:
Nit: What if I'm on Windows?

In the build or start commands, Instead of `sh /script.sh` could we do `chmod +x ./script && sh ./script.sh`

Reply:
Oh, that's smart. So instead of doing it locally, the server does that for us when it's about to invoke it? 

But wouldn't that be repetitive? because it means every time we deploy, it always has to run the change permission command. that seems weird 🤔

Reply:
Yeah, but I would rather that than answer a support thread on why the script doesn't have execution permissions.